src/main.rs

```diff
@@ -153,24 +124,43 @@ fn main() -> xcb::Result<()> {
 
 	// It is now time to finalize the initialization of AquariWM by instantiating the main window
 	// manager.
-	let wm = AquariWm::new(conn, root);
-	// TODO: Explore possible options for making this cleaner/more idiomatic. Ideally,
-	//       instantiating the window manager and running the event loop would be the same
-	//       function call.
-	wm.run()
+	AquariWm::start(conn, root)?;
+	Ok(())
 }
 
 /// Initializes the given [window](x::Window) by requesting to receive certain events on it.
 ///
-/// Requests to receive the following events:
+/// Grabs all button presses, releases and drags when the Super key (mod mask 4) is held, and
+/// registers for the following events:
 /// - [`ENTER_WINDOW`](x::EventMask::ENTER_WINDOW)
 /// - [`FOCUS_CHANGE`](x::EventMask::FOCUS_CHANGE)
 fn init_window(conn: &Connection, window: &Window) -> xcb::Result<()> {
-	trace!("Initializing window ({})", window.resource_id());
+	trace!(
+		window = window.resource_id(),
+		"Grabbing mouse buttons on window"
+	);
+	conn.send_request(&x::GrabButton {
+		owner_events: false,
+		grab_window: *window,
+		event_mask: x::EventMask::BUTTON_PRESS
+			| x::EventMask::BUTTON_RELEASE
+			| x::EventMask::BUTTON_MOTION,
```

Comment:
If changing to not require Super to be continually held, this grab should only be for `BUTTON_PRESS`. After receiving a button press event and starting window manipulation, an additional button grab should be made for the `BUTTON_RELEASE` and `BUTTON_MOTION` masks.